corehq/apps/tour/models.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+from django.contrib.auth.models import User
+from django.db import models
+import json_field
+
+
+class GuidedTours(models.Model):
+    user = models.ForeignKey(
+        User,
+        on_delete=models.PROTECT,
+        unique=True
+    )
+    seen_tours = json_field.JSONField(
+        default={},
+    )
+
+    @classmethod
+    def has_seen_tour(cls, user, tour_slug):
+        guided_tour, _ = GuidedTours.objects.get_or_create(user=user)
+        return tour_slug not in guided_tour.seen_tours
```

Comment:
Hmm, lot of things happening here: SELECT, (possible INSERT), (possible SELECT, depending on Django's implementation), deserialize object, deserialize JSON. This process could be greatly simplified with a different table structure:
```py
class GuidedTours(models.Model):
    user = models.ForeignKey(User, index=True, ...)
    slug = models.String(index=True)

    @classmethod
    def has_seen_tours(cls, user, tour_slug):
        return cls.object.filter(user=user, slug=tour_slug).count() > 0

    @classmethod
    def mark_as_seen(cls, user, tour_slug):
        cls.objects.get_or_create(user=user, slug=tour_slug)
```